plugins/optimization-detective/site-health.php

```diff
@@ -0,0 +1,171 @@
+		$result['status']      = 'recommended';
+		$result['label']       = __( 'Error accessing the REST API endpoint', 'optimization-detective' );
+		$result['description'] = esc_html__( 'There was an issue reaching the REST API endpoint. This might be due to server settings or the REST API being disabled.', 'optimization-detective' );
+		$info                  = array(
+			'error_message' => $result['description'],
+			'error_code'    => $response->get_error_code(),
+			'available'     => false,
+		);
+	} else {
+		$status_code     = wp_remote_retrieve_response_code( $response );
+		$data            = json_decode( wp_remote_retrieve_body( $response ), true );
+		$expected_params = array( 'slug', 'current_etag', 'hmac', 'url', 'viewport', 'elements' );
+		$info            = array(
+			'status_code' => $status_code,
+			'available'   => false,
+		);
+
+		if (
+			400 === $status_code
+			&& isset( $data['data']['params'] )
+			&& is_array( $data['data']['params'] )
+			&& count( $expected_params ) === count( array_intersect( $data['data']['params'], $expected_params ) )
+		) {
+			// The REST API endpoint is available.
+			$info['available'] = true;
+		} elseif ( 401 === $status_code ) {
+			$result['status']      = 'recommended';
+			$result['label']       = __( 'Authorization should not be required to access the REST API endpoint.', 'optimization-detective' );
+			$result['description'] = esc_html__( 'To collect URL metrics, the REST API endpoint should be accessible without requiring authorization.', 'optimization-detective' );
+		} elseif ( 403 === $status_code ) {
+			$result['status']      = 'recommended';
+			$result['label']       = __( 'The REST API endpoint should not be forbidden.', 'optimization-detective' );
+			$result['description'] = esc_html__( 'The REST API endpoint is blocked. Please review your server or security settings.', 'optimization-detective' );
+		} else {
+			$result['status']      = 'recommended';
+			$result['label']       = __( 'Error accessing the REST API endpoint', 'optimization-detective' );
+			$result['description'] = esc_html__( 'There was an issue reaching the REST API endpoint. This might be due to server settings or the REST API being disabled.', 'optimization-detective' );
+		}
+		$info['error_message'] = $result['description'];
+	}
+
+	$result['description'] = sprintf(
+		'<p>%s</p>',
+		$result['description']
+	);
+
+	update_option( 'od_rest_api_info', $info );
+	return $result;
+}
+
+/**
+ * Displays an admin notice if the REST API health check fails.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $plugin_file Plugin file.
+ */
+function od_rest_api_health_check_admin_notice( string $plugin_file ): void {
+	if ( 'optimization-detective/load.php' !== $plugin_file ) {
+		return;
+	}
+
+	$rest_api_info = get_option( 'od_rest_api_info', array() );
+	if (
+		isset( $rest_api_info['available'] ) &&
+		false === $rest_api_info['available'] &&
+		isset( $rest_api_info['error_message'] )
```

Comment:
Based on my above suggestion, this would need to become:
```suggestion
		isset( $rest_api_info['error_message'] ) &&
		'' !== $rest_api_info['error_message']
```